### ISSUE TITLE
no-conditional-assignment doesn't warn in ternary operator

### ISSUE THREAD

**Issue by fedeolto** [palantir/tslint#4837]
### Bug Report

- __TSLint version__: 5.19.0
- __TypeScript version__: 3.4.5
- __Running TSLint via__: (pick one) CLI / Visual Studio Code/

#### TypeScript code being linted

```ts
const action = this.sortOrder = 'asc' ? 'running' : 'sitting';
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint-angular"],
  "rules": {
    "max-line-length": [
      true, 
      {
        "limit": 180, 
        "ignore-pattern": "^import |^export | implements | it"
      }
    ],
    "no-conditional-assignment": true,
    "no-sparse-arrays": true
  } 
}
```

#### Actual behavior
the line causes no warning nor errors. 

If I add the following lines:
```ts
if (this.sortOrder = 'ciao') {
  console.log('miao');
}
```
I obtain the error: " Assignments in conditional expressions are forbidden"

Doc says: 
"This applies to do-while, for, if, and while statements and conditional (ternary) expressions."
So I expect that a an assignment in a ternary operator would have raised an error

#### Expected behavior
An error stating that there's an assignment in a conditional expression

**Comment by fatcerberus** [comment]
Both assignment and the ternary operator are right associative, so I'm fairly certain this is parsed as:

```js
const action = this.sortOrder = ('asc' ? 'running' : 'sitting');
```

and not:

```js
const action = (this.sortOrder = 'asc') ? 'running' : 'sitting';
```

Therefore there is no assignment in a conditional here.  It's simply a chained assignment.

**Comment by fedeolto** [comment]
Thanks for the pointing out what happens behind the scenes

I have two questions:
1) What would be a scenario in which the rule complains about "conditional (ternary)" expressions as stated in the documentation?
2) Is there any rule that marks `x = y = true ? 'pig' : 'cow'` as a possible flaw?

**Comment by fatcerberus** [comment]
1. This will probably trigger the error:
```ts
const action = (this.sortOrder = 'asc') ? 'running' : 'sitting';
```

2. Chained assignment is a fairly common pattern: e.g. `cur = defValue = 0`.  The actual flaw there is that the conditional part `true ?` is constant; I don’t think there’s a core rule to catch that.

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
